lib/matplotlib/backend_tools.py

```diff
@@ -626,28 +625,44 @@ def scroll_zoom(self, event):
         # https://gist.github.com/tacaswell/3144287
         if event.inaxes is None:
             return
-        ax = event.inaxes
-        cur_xlim = ax.get_xlim()
-        cur_ylim = ax.get_ylim()
-        # set the range
-        cur_xrange = (cur_xlim[1] - cur_xlim[0])*.5
-        cur_yrange = (cur_ylim[1] - cur_ylim[0])*.5
-        xdata = event.xdata  # get event x location
-        ydata = event.ydata  # get event y location
+
+        scl = self.base_scale
+
         if event.button == 'up':
             # deal with zoom in
-            scale_factor = 1 / self.base_scale
+            direction = 'in'
         elif event.button == 'down':
             # deal with zoom out
-            scale_factor = self.base_scale
+            direction = 'out'
         else:
             # deal with something that should never happen
-            scale_factor = 1
-        # set new limits
-        ax.set_xlim([xdata - cur_xrange*scale_factor,
-                     xdata + cur_xrange*scale_factor])
-        ax.set_ylim([ydata - cur_yrange*scale_factor,
-                     ydata + cur_yrange*scale_factor])
+            scl = 1
+
+        ax = event.inaxes
+        # get the bounding box
+        transD2C = ax.transData.transform
+        xmin, xmax = ax.get_xlim()
+        ymin, ymax = ax.get_ylim()
```

Comment:
Why don't you use `ax._get_view()` here?
The idea is for this tool to be compatible with all kind of axes that change the view limits.